src/Avalonia.Base/Animation/Animation.cs

```diff
@@ -248,7 +251,11 @@ public static void SetAnimator(IAnimationSetter setter, Type value)
         /// <typeparam name="TAnimator">
         /// The type of the animator to instantiate.
         /// </typeparam>
-        public static void RegisterAnimator<TAnimator>(Func<AvaloniaProperty, bool> condition)
+        public static void RegisterAnimator<
```

Comment:
IMO we should just use `new()` constraint and register `Func<IAnimator>` instead of a type. That would skip reflection entirely and be compatible with reflectionless mode later

Reply:
The `new()` constraint is still using `Activator.CreateInstance` to create instances. 
If we want to avoid reflection completely, we should consider using the new language feature `Abstract Static Methods in Interfaces` which is only available in .NET 6+.

Reply:
Yes, but Activator.CreateInstance<T> it's automatically trimmable and is compatible with reflection-free NativeAOT mode.

Reply:
I think we can make use of abstract static interface method given that nativeaot only available on .NET 6+